pkg/controller/risingwave_controller.go

```diff
@@ -364,8 +368,16 @@ func (c *RisingWaveController) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(c)
 }
 
-func NewRisingWaveController(client client.Client) *RisingWaveController {
+func NewRisingWaveController(client client.Client, recorder record.EventRecorder) *RisingWaveController {
+	rw := &risingwavev1alpha1.RisingWave{}
+	hookFactory := func() ctrlkit.ActionHook { return NewEventHook(recorder, rw) }
+	callback := func(risingWave *risingwavev1alpha1.RisingWave) {
+		rw = risingWave
+	}
```

Comment:
I think it's more intuitive to capture the risingwave directly with closure instead of a callback. 

We can have a concept like `ChainedActionHook` to wrap a list of hooks into one, and then we can inject this event hook by enhancing the `managerOpts() -> managerOpts(risingwave)`.